internal/util/fencing.go

```diff
@@ -0,0 +1,142 @@
+package util
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/csi-addons/spec/lib/go/fence"
+	"github.com/golang/protobuf/ptypes/timestamp"
+)
+
+type NetworkFence struct {
+	Cidr      string
+	Monitors  string
+	CreatedAt *timestamp.Timestamp
+}
+
+// addCephBlocklist adds an IP to ceph osd blocklist.
+func (nwFence *NetworkFence) addCephBlocklist(ctx context.Context, ip string, cr *Credentials) error {
+	arg := []string{
+		"--id", cr.ID,
+		"--keyfile=" + cr.KeyFile,
+		"-m", nwFence.Monitors,
+	}
+	cmd := []string{"osd", "blocklist", "add"}
+	cmd = append(cmd, arg...)
+
+	_, stderr, err := ExecCommand(ctx, "ceph", cmd...)
+	if err != nil {
+		return fmt.Errorf("failed to blocklist IP %s: %v", ip, stderr)
+	}
+
+	return nil
+}
+
+// AddNetworkFence blocks access for all the IPs in the IP range mentioned via the CIDR block
+// using a network fence.
+func AddNetworkFence(ctx context.Context, nwFence *NetworkFence, cr *Credentials) error {
```

Comment:
make this as method of NetworkFence struct

Reply:
Done :+1: